### PR TITLE
Add file getMetadata & getFileasArray operation

### PR DESCRIPTION
**Pull request by SanduDS**
# Description

1. Adding get file metadata operation
2. Adding get a file as a byte array opearation

Fixes https://github.com/ballerina-platform/ballerina-extended-library/issues/315, https://github.com/ballerina-platform/ballerina-extended-library/issues/311

One line release note: 
- Adding `get file metadata `and` get file as a byte array` operations 

## Type of change
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
All tests passed

**Test Configuration**:
* Ballerina Version: Swan Lake 2201.0.1
* Operating System: Ubuntu
* Java SDK: 11

# Checklist:

### Security checks
 - [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: storageservice/modules/files/file_client.bal [hunk 1]
```diff
@@ -78,7 +78,7 @@ public isolated client class FileClient {
         }
         map<string> headerMap = populateHeaderMapFromRequest(request);
         http:Response response = <http:Response>check self.httpClient->get(requestPath, headerMap);
-        if (response.statusCode === http:STATUS_OK ) {
+        if (response.statusCode === http:STATUS_OK) {
```
Comment (abeykoon) [inline storageservice/modules/files/file_client.bal:81]: what is the reason to use `===`? We have `==` also in Ballerina right?

File: storageservice/modules/files/file_client.bal [hunk 2]
```diff
@@ -240,6 +240,52 @@ public isolated client class FileClient {
...
+    # + fileName - Name of the File  
+    # + azureDirectoryPath - Path of the azure directory
+    # + uriParameters - Optional URI Parameter Description
+    # + return - If successful, File Metadata. Else an Error
+    @display {label: "Get File Metadata"}
+    remote isolated function getFileMetadata(@display {label: "File Share Name"} string fileShareName,
+                                             @display {label: "File Name"} string fileName, 
+                                             @display {label: "Azure Directory Path"}string? azureDirectoryPath = (),
+                                             @display {label: "Optional Parameters"} OptionalURIParametersFileMetaData
+                                             uriParameters = {}) returns @display {label: "File metadata"}
```
Comment (abeykoon) [inline storageservice/modules/files/file_client.bal:255]: `OptionalURIParametersFileMetaData ` does this get properly rendered in Choreo lowcode?

File: storageservice/modules/files/file_client.bal [hunk 3]
```diff
@@ -384,6 +430,49 @@ public isolated client class FileClient {
...
+    # + fileShareName - Name of the FileShare
+    # + fileName - Name of the file
+    # + azureDirectoryPath - Path of azure directory
+    # + return -  If successful, file content as a byte array. Else an error
+    @display {label: "Download File"}
+    remote isolated function getFileAsByteArray(@display {label: "File Share Name"} string fileShareName, 
+                                     @display {label: "Azure File Name"} string fileName,  
+                                     @display {label: "Azure Directory Path"} string? azureDirectoryPath = ()) 
+                                     returns @display {label: "Response"} byte[]|error {
+        string requestPath = azureDirectoryPath is () ? (SLASH + fileShareName + SLASH + fileName) : (SLASH
```
Comment (abeykoon) [inline storageservice/modules/files/file_client.bal:445]: File separator? Doesn't Ballerina provide a const for this?

File: storageservice/modules/files/file_client.bal [hunk 4]
```diff
@@ -384,6 +430,49 @@ public isolated client class FileClient {
...
+                resourcePath: resourcePathForSharedkeyAuth,
+                requiredURIParameters: requiredURIParameters
+            };
+            check prepareAuthorizationHeaders(authorizationDetail);      
+        } else {
+            requestPath = requestPath.concat(QUESTION_MARK, self.azureConfig.accessKeyOrSAS.substring(1)); 
+        }
+        map<string> headerMap = populateHeaderMapFromRequest(request);
+        http:Response response = <http:Response> check self.httpClient->get(requestPath, headerMap);
+        if (response.statusCode === http:STATUS_OK ) {
```
Comment (abeykoon) [inline storageservice/modules/files/file_client.bal:465]: Do we need this check? If a response other than `http:STATUS_OK` is returned, it is interpreted as an error and will be handled by `check`. Is my understanding wrong there?

File: storageservice/modules/files/file_client.bal [hunk 5]
```diff
@@ -384,6 +430,49 @@ public isolated client class FileClient {
...
+    @display {label: "Download File"}
+    remote isolated function getFileAsByteArray(@display {label: "File Share Name"} string fileShareName, 
+                                     @display {label: "Azure File Name"} string fileName,  
+                                     @display {label: "Azure Directory Path"} string? azureDirectoryPath = ()) 
+                                     returns @display {label: "Response"} byte[]|error {
+        string requestPath = azureDirectoryPath is () ? (SLASH + fileShareName + SLASH + fileName) : (SLASH 
+            + fileShareName + SLASH + azureDirectoryPath + SLASH + fileName);    
+        http:Request request = new;
+        if (self.azureConfig.authorizationMethod === ACCESS_KEY) {
+            map<string> requiredURIParameters ={};
```
Comment (LakshanSS) [inline storageservice/modules/files/file_client.bal:449]:
```suggestion
            map<string> requiredURIParameters = {}; 
```

File: storageservice/modules/files/file_client.bal (hunk 1 above)
Comment (SanduDS) [reply storageservice/modules/files/file_client.bal:81]: This connector was initially implemented using older ballerina swan lake versions. There are places that we can optimize and improve with the latest version. I will work on this.

File: storageservice/modules/files/file_client.bal (hunk 3 above)
Comment (BuddhiWathsala) [reply storageservice/modules/files/file_client.bal:445]: @kalaiyarasiganeshalingam do we support this in `file` module?

Comment (kalaiyarasiganeshalingam) [reply storageservice/modules/files/file_client.bal:445]: No, we don't have any public constant for this in the file module.

File: storageservice/modules/files/file_client.bal (hunk 2 above)
Comment (SanduDS) [reply storageservice/modules/files/file_client.bal:255]: Yes, this is similar to the other function's option URL parameter inputs records

Comment (SanduDS) [reply storageservice/modules/files/file_client.bal:255]: image

**Review by LakshanSS — APPROVED** [review]

**Pull request merged**